### ISSUE TITLE
Investigate whether Incremental sync - deduped history mode can be supported

### ISSUE THREAD

**Issue by orware** [planetscale/airbyte-source#86]
There may be an underlying architectural reason as to why we may not be able to support this within our Airbyte source connector, but I wanted to file an official issue requesting whether the "Incremental sync - deduped history" mode could be supported on behalf of several of our users who have indicated they've been trying to do so with their destination connectors and have observed some issues.

A recent case has shared that they are experiencing the same issue as described in https://github.com/planetscale/airbyte-source/issues/56 and also observing that not all updates seem to be coming across properly on subsequent syncs when attempting to use the "Incremental sync - deduped history" mode. 

The user has been encouraged to utilize the "Incremental sync - append" mode instead which seems to be working well enough, but they are still curious about the deduped history option as well as a solution for the `ENUM` value issue.

Here are some details provided by a user:

> [This] is a critical concern for us because we are not able to get data replicated properly to our destination database redshift. Just to explain further when we create a connection with below configuration, only new inserts are synced to the destination. None of the updates gets synced to destination.
> 
> Normalization: Normalized tabular data (no custom transformers)
> Sync mode: Incremental | Append + Deduped
> 
> Some of the updates are synced properly. But there are a large number of updates that are not replicated. I am also providing an example transaction below for your understanding.
> 
> **Source - PlanetScale**
> 
> "id" is the primary key in source and this is by default selected as the cursor field and primary key while configuring the Airbyte connection.
> 
> id | created_at | updated_at | status
> -- | -- | -- | --
> example_p00b7md0f9fts5b6m | "2023-10-12 00:00:03.878" | "2023-10-12 00:00:27.888" | "success"
> 
> **Destination - Redshift**
> 
> If you look at the destination record, in `status` field we have value 1, which actually corresponds to the numeric index of the value "pending". You can also see that `updated_at` is still not modified. It is still having the value when the record was inserted at source. Status should have been updated to "success" or at least to "2" (2 corresponds to the numeric index of value success).
> 
> id | created_at | updated_at | status
> -- | -- | -- | --
> example_p00b7md0f9fts5b6m | "2023-10-12 00:00:03.878" | "2023-10-12 00:00:03.878" | "1"

**Comment by majgaard** [comment]
**Disclaimer**: I'm not a contributor to this repository and my familiarity with the inner workings of Vitess is very limited.

From what I can gather from the source code -- it seems like there's no reference to the _dedupe_ functionality of AirByte in this repository. I believe it's a downstream feature of AirByte (and implemented on the destination connector).

It looks like for the first sync to AirByte (where no cursor exists yet), that [VStream initiates a table-copy phase](https://github.com/vitessio/vitess/issues/6277). It's clear from my own trials (and the behavior mentioned in this issue & in #56) that Enums import just fine on first sync.

Subsequent syncs (which actually use a VGtid as the cursor) rely on the binlog. [This .proto file](https://github.com/vitessio/vitess/blob/43e533bdb7ada66e864dddccb8b3024e8448004e/proto/binlogdata.proto#L150C1-L151C60) makes reference to enums in the binlog being numeric. Could it be that binlog events sent to AirByte which impact enum fields just aren't being properly enriched with the corresponding value?

**Question**: Does this repository make use of the [FIELD event](https://github.com/vitessio/vitess/issues/6955)'s enum definitions?

**Another PlanetScale & VStream related question I haven't been able to find the answer to**: What's the maximum amount of binlog PlanetScale allows users to stack up? (i.e. How many GBs of new / changed data can I have laying around between AirByte syncs before PlanetScale begins to prune the history?) Please do let me know if this is a dumb question -- I simply couldn't find the answer myself.

**Comment by notfelineit** [comment]
@majgaard You're correct about how `ENUM` values are treated differently during the table-copy phase vs. the replication phase ("subsequent syncs"). We're putting in a fix based on that exact behavior, that uses the FIELD `columnType` 🙂
